components/director/internal/domain/application/service.go

```diff
@@ -82,20 +88,22 @@ type UIDService interface {
 }
 
 type service struct {
-	appRepo      ApplicationRepository
-	apiRepo      APIRepository
-	eventAPIRepo EventAPIRepository
-	documentRepo DocumentRepository
-	webhookRepo  WebhookRepository
-	labelRepo    LabelRepository
-	runtimeRepo  RuntimeRepository
+	appRepo          ApplicationRepository
+	apiRepo          APIRepository
+	eventAPIRepo     EventAPIRepository
+	documentRepo     DocumentRepository
+	webhookRepo      WebhookRepository
+	labelRepo        LabelRepository
+	runtimeRepo      RuntimeRepository
+	fetchRequestRepo FetchRequestRepository
 
 	labelUpsertService LabelUpsertService
 	scenariosService   ScenariosService
 	uidService         UIDService
+	timestampGen       timestamp.Generator
 }
 
-func NewService(app ApplicationRepository, webhook WebhookRepository, api APIRepository, eventAPI EventAPIRepository, documentRepo DocumentRepository, runtimeRepo RuntimeRepository, labelRepo LabelRepository, labelUpsertService LabelUpsertService, scenariosService ScenariosService, uidService UIDService) *service {
+func NewService(app ApplicationRepository, webhook WebhookRepository, api APIRepository, eventAPI EventAPIRepository, documentRepo DocumentRepository, runtimeRepo RuntimeRepository, labelRepo LabelRepository, fetchRequestRepo FetchRequestRepository, labelUpsertService LabelUpsertService, scenariosService ScenariosService, uidService UIDService) *service {
```

Comment:
Just wondering whether it may be nice to wrap those Repository parameters with some encapsulating object?

Reply:
I will create issue for that